mobileRobot:
  # Publish all joint states
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50
  
  # Effort controller 
  left_b_effort_controller:
    type: effort_controllers/JointEffortController
    joint: left_b
  left_m_effort_controller:
    type: effort_controllers/JointEffortController
    joint: left_m
  left_f_effort_controller:
    type: effort_controllers/JointEffortController
    joint: left_f
  right_b_effort_controller:
    type: effort_controllers/JointEffortController
    joint: right_b
  right_m_effort_controller:
    type: effort_controllers/JointEffortController
    joint: right_m
  right_f_effort_controller:
    type: effort_controllers/JointEffortController
    joint: right_f

  # Velocity controller 
  left_b_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: left_b
    pid: {p: 100.0, i: 0.01, d: 10.0}
  left_m_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: left_m
    pid: {p: 100.0, i: 0.01, d: 10.0}
  left_f_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: left_f
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_b_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: right_b
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_m_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: right_m
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_f_velocity_controller:
    type: effort_controllers/JointVelocityController
    joint: right_f
    pid: {p: 100.0, i: 0.01, d: 10.0}
